### PR TITLE
Text index indexing optimizations

### PR DESCRIPTION
**Pull request by jotare**
### Description
Small optimizations to avoid unneeded `clone`s and some renaming

### How was this PR tested?
Describe how you tested this PR.

### CODE REVIEW

File: nucliadb_texts/src/writer.rs [hunk 1]
```diff
@@ -225,35 +231,31 @@ impl TextWriterService {
...
-        for label in resource_labels {
+        for label in resource.labels.iter() {
             let facet = Facet::from(label.as_str());
-            doc.add_facet(self.schema.facets, facet);
+            base_doc.add_facet(self.schema.facets, facet);
         }
 
         for (field, text_info) in &resource.texts {
-            let mut subdoc = doc.clone();
+            let mut doc = base_doc.clone();
```
Comment (hermeGarcia) [inline nucliadb_texts/src/writer.rs:247]: Maybe call it `new_doc` instead of just `doc` to not create confusion with the macro.

**Review by hermeGarcia — CHANGES_REQUESTED** [review]
LGTM, just one detail

File: nucliadb_texts/src/writer.rs (hunk 1 above)
Comment (jotare) [reply nucliadb_texts/src/writer.rs:247]: What about `field_doc`, as it's a doc for a field?

Comment (hermeGarcia) [reply nucliadb_texts/src/writer.rs:247]: sounds good

**Review by hermeGarcia — APPROVED** [review]

**Pull request merged**